### DISCUSSION TITLE
Why docker sends SIGKILL after 10 seconds ?

### DISCUSSION THREAD

**Discussion by Fomys** [moby/moby#45101] [Q&A] [answered]
Hello,
This may be a stupid question, but I can't find the answer anywhere.
Why does `docker stop` send SIGKILL after 10 seconds?

Because when I do a `docker stop` I expect docker to pass a SIGTERM signal to the process and wait for it to finish. In particular, docker should wait for the process to finish running to allow a proper data save. Because of this illogical behavior, I just lost an hour's worth of data because docker didn't give the process time to save properly (yes, I have slow storage, but 10 seconds is very short...).
This doesn't really make sense to me, especially since there is also the `docker kill` command which is designed to force the process to terminate.

I found a way to stop this from happening to me by setting a very long stop_grace_period on this particular docker (https://docs.docker.com/compose/compose-file/#stop_grace_period), but is there any way to configure docker globally so that it never sends this SIGKILL?

Thank you very much

**Comment by cpuguy83** [comment, marked as answer]
`docker stop` stops the container. It does so by passing along the configured termination signal and if it doesn't stop it forces it to.
In terms of "why"? I'm not sure I or likely anyone can give you the real answer here, it's been that way probably since it was first created... certainly since I started using docker pre-1.0 and works very well most of the time.
You can pass `-t -1` to make it wait indefinitely or specific some other longer timeout.

Also worth noting `docker kill` just sends a signal to the container, and defaults to `SIGKILL`.
You could use `docker kill -s SIGTERM && docker wait`  as well.

**Reply by a3gh** [reply]
This is also possible with docker compose: https://docs.docker.com/reference/compose-file/services/#stop_grace_period
